libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java

```diff
@@ -960,15 +960,11 @@ protected void addAccount(final UserAccount account) {
                 accountOptions.csrfToken);
 
     	/*
-    	 * Registers for push notifications, if push notification client ID is present.
+    	 * Registers for push notifications, if setup by the app.
     	 * This step needs to happen after the account has been added by client
     	 * manager, so that the push service has all the account info it needs.
     	 */
-        final Context appContext = SalesforceSDKManager.getInstance().getAppContext();
-        final String pushNotificationId = BootConfig.getBootConfig(appContext).getPushNotificationClientId();
-        if (!TextUtils.isEmpty(pushNotificationId)) {
-            PushMessaging.register(appContext, account);
-        }
+        PushMessaging.register(SalesforceSDKManager.getInstance().getAppContext(), account);
```

Comment:
Is there any reason we can think of to pass context in here instead of just having it in the object?  It is passed to just about ever function in `PushMessaging` and that just feels ugly/unnecessary.

Reply:
Not really.

Reply:
Static references to Android Context are a memory leak 😞